Makefile

```diff
@@ -1,60 +1,28 @@
-GTEST_DIR ?= external/googletest/googletest
-CXX ?= clang++
-CXXFLAGS += -std=c++14 -pedantic -Wall -Wextra -Iinclude/
-CC ?= clang
-CFLAGS += -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter -Iinclude/
-AR ?= ar
-RANLIB ?= ranlib
-XCODEBUILD ?= xcodebuild
+BUILD := build
+CMAKE := cmake
+MAKE := make
 
-SOURCES = $(shell find src -name '*.cpp')
-OBJECTS = $(SOURCES:.cpp=.o)
-LIBRARY = libArbiter.a
+all: cmake
+	$(MAKE) -C $(BUILD)
 
-TEST_SOURCES = $(shell find test -name '*.cpp') $(GTEST_DIR)/src/gtest-all.cc $(GTEST_DIR)/src/gtest_main.cc
-TEST_RUNNER = test/main
-TEST_INCLUDES = -isystem $(GTEST_DIR)/include -I$(GTEST_DIR) -Isrc/
+cmake:
+	mkdir -p $(BUILD)
+	$(CMAKE) -B$(BUILD) -H. -G "Unix Makefiles" -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE
```

Comment:
I'd like to make this Makefile check for the presence of `ninja` and use that if available.